Source/Core/VideoBackends/Software/SWTexture.h

```diff
@@ -23,6 +23,15 @@ class SWTexture final : public AbstractTexture
                                 const MathUtil::Rectangle<int>& dstrect) override;
   void Load(u32 level, u32 width, u32 height, u32 row_length, const u8* buffer,
             size_t buffer_size) override;
+
+  u8* GetData() const;
+  size_t GetDataSize() const;
+
+private:
+  std::optional<RawTextureInfo> MapFullImpl() override;
+
+  u8* m_data = nullptr;
```

Comment:
This should likely be a `std::vector`, which would also get rid of the separate data size variable.

`Load` will then simplify from:

```cpp
void SWTexture::Load(u32 level, u32 width, u32 height, u32 row_length, const u8* buffer,
                     size_t buffer_size)
{
  if (m_data)
  {
    delete[] m_data;
  }
  m_data = new u8[buffer_size];
  memcpy(m_data, buffer, buffer_size);
  m_data_size = buffer_size;
}
```

to

```cpp
void SWTexture::Load(u32 level, u32 width, u32 height, u32 row_length, const u8* buffer,
                     size_t buffer_size)
{
  m_data.assign(buffer, buffer + buffer_size);
}
```